### ISSUE TITLE
Unsupported metric data type ExponentialHistogram in opentelemetry-exporter-gcp-monitoring

### ISSUE THREAD

**Issue by matthieu-dujany-technis** [GoogleCloudPlatform/opentelemetry-operations-python#357]
Hello,

I am having issues exporting an opentelemetry histogram to Google Cloud with **exponential bucketing.**

I get the error "Unsupported metric data type ExponentialHistogram, ignoring it" which I believe comes from this [code](https://github.com/GoogleCloudPlatform/opentelemetry-operations-python/blob/main/opentelemetry-exporter-gcp-monitoring/src/opentelemetry/exporter/cloud_monitoring/__init__.py) line 216.

I am using the following code:

`from opentelemetry.exporter.cloud_monitoring import (
    CloudMonitoringMetricsExporter,
)
from opentelemetry.sdk.metrics import MeterProvider, Histogram
from opentelemetry.sdk.metrics.export import PeriodicExportingMetricReader
from opentelemetry.sdk.resources import Resource
from opentelemetry.sdk.metrics.view import View
import random
import time
from opentelemetry.sdk.metrics._internal.aggregation import (
    ExponentialBucketHistogramAggregation,
)

project_id = "my_project_id"

meter_provider = MeterProvider(
    metric_readers=[
        PeriodicExportingMetricReader(
            CloudMonitoringMetricsExporter(
                project_id=project_id,
            ),
            export_interval_millis=30000,
        )
    ],
    resource=Resource.create(
        {
            "service.name": "test-matthieu-opentelemetry",
        }
    ),
    views=[
        View(
            instrument_type=Histogram,
            aggregation=ExponentialBucketHistogramAggregation(),
        )
    ],
)

meter = meter_provider.get_meter("test-meter")
request_duration_seconds = meter.create_histogram("test_histogram")

while True:
    value = random.random()
    request_duration_seconds.record(value)
    time.sleep(10)
`

with the following packages on Python 3.10.12:
opentelemetry-api==1.27.0
opentelemetry-exporter-gcp-monitoring==1.7.0a0
opentelemetry-resourcedetector-gcp==1.7.0a0
opentelemetry-sdk==1.27.0

If I remove the view part with the ExponentialBucketHistogramAggregation, the export then runs successfully but with the default explicit histogram bucketing boundaries defined [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#explicit-bucket-histogram-aggregation). I would like to be able to run it with the ExponentialBucketHistogramAggregation.

I have seen that is possible to do it in opentelemetry-operations-js with this example in the [README](https://github.com/GoogleCloudPlatform/opentelemetry-operations-js/blob/main/packages/opentelemetry-cloud-monitoring-exporter/README.md#enabling-exponential-histograms).

Is there something that I am missing ? Or are exponential buckets not supported in opentelemetry-operations-python (as the error seems to indicate) ?

Thanks for the help,

**Comment by aabmass** [comment]
> Is there something that I am missing ? Or are exponential buckets not supported in opentelemetry-operations-python (as the error seems to indicate) ?

That's correct, its just not implemented yet. It should be pretty straightforward to add though, the logic can be copied from the [Go](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/d6795b491c748855e4a4d6eec07ae2cb44a78da1/exporter/metric/metric.go#L723) or [JS](https://github.com/GoogleCloudPlatform/opentelemetry-operations-js/blob/81b32a1c23a0a0b1129b555c7db2a4c2c8d8e672/packages/opentelemetry-cloud-monitoring-exporter/src/transform.ts#L259) exporters where it is already implemented.

I'm going to mark this as up for grabs as I can't get to it right now. If you or anyone else wants to take a shot, let me know and I can assign it to you 🙂

**Comment by tgwizard** [comment]
Fix here: https://github.com/GoogleCloudPlatform/opentelemetry-operations-python/pull/413
